ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -84,22 +103,263 @@ where
     Ok(changed_keys)
 }
 
+fn act_on_transfers_to_eth<D, H>(
+    storage: &mut Storage<D, H>,
+    transfers: &[TransferToEthereum],
+) -> Result<BTreeSet<Key>>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let mut changed_keys = BTreeSet::default();
+    // all keys of pending transfers
+    let prefix = BRIDGE_POOL_ADDRESS.to_db_key().into();
+    let mut pending_keys: HashSet<Key> = storage
+        .iter_prefix(&prefix)
+        .0
+        .map(|(k, _, _)| {
+            Key::from_str(k.as_str()).expect("Key should be parsable")
+        })
+        .filter(is_pending_transfer_key)
+        .collect();
+
+    // Remove the completed transfers from the bridge pool
+    for event in transfers {
+        let pending_transfer = event.into();
+        let key = get_pending_key(&pending_transfer);
+        if storage.has_key(&key)?.0 {
+            _ = storage.delete(&key)?;
+            _ = pending_keys.remove(&key);
+        } else {
+            return Err(eyre::eyre!(
+                "The transfer doesn't exist in the bridge pool"
+            ));
```

Comment:
I think the only way we could have run into this code path is if a quorum of validators decided on an event that didn't happen (that is, effectively, we would have had a Byzantine quorum of validators at that point). to transfer to Ethereum, users need to provide a proof that they sent a transfer to the bridge pool, at which point their transfer would be registered in storage.

therefore, maybe it's wiser to panic here? or use `unreachable!()`? there's also `namada_core::hints::likely()`, for the optimistic if branch.